postprocessing/science/get_distances_to_rupture.jl

```diff
@@ -0,0 +1,216 @@
+# Script to calculate precise distances from SeisSol's surface/volume cells to the rupturing part of the fault
+# Output will be saved as a numpy file (.npy)
+# Multithreading in julia can be enabled with the flag -t
+# Example start command: mpiexecjl -np 5 julia -t 10 get_distances_to_rupture.jl data-surface.xdmf
+
+using SeisSolXDMF
+using ArgParse
+using NPZ
+
+function parse_commandline()
+    s = ArgParseSettings()
+    @add_arg_table s begin
+        "--slip_threshold"
+            help = "slip threshold to define the rupturing part of the fault"
+            arg_type = Float32
+            default = 0.1f0
+        "--use_gpu"
+            help = "enable GPU acceleration with CUDA.jl"
+            action = :store_true
+        "--timestep_range"
+            help = "parse the timestep range that determines the rupturing part of the fault \
+                    (e.g., needed if the output file contains more than one event)"
+            nargs = 2
+            arg_type = Int
+            default = [1,-1]
+        "--noMPI"
+            help = "disable MPI; this flag will increase the performance of a single rank"
+            action = :store_true
+        "--joyner_boore_distance"
+            help = "compute Joyner-Boore distance instead of rupture distance (ignoring topography)"
+            action = :store_true
+        "filename"
+            help = "path to SeisSol's surface or volume.xdmf"
+            required = true
+    end
+    return parse_args(s)
+end
+
+
+const parsed_args = parse_commandline()
+const filename = parsed_args["filename"]
+const threshold = parsed_args["slip_threshold"]
+const timestep_range = parsed_args["timestep_range"]
+const use_gpu = parsed_args["use_gpu"]
+const noMPI = parsed_args["noMPI"]
+const joyner_boore_distance = parsed_args["joyner_boore_distance"]
+
+
+function get_cell_barycenters(xdmf)
+    topology, geometry = grid_of(xdmf)
+    xyz = (1. / 3.) .* (geometry[:,topology[1,:]] .+ geometry[:,topology[2,:]] .+ geometry[:,topology[3,:]])
+    return copy(transpose(convert(Matrix{Float32}, xyz)))
+end
+
+
+function get_ASl(xdmf, timestep_range::Vector{Int})
+    
+    ntimesteps = size(timesteps_of(xdmf))[1]
+    if !(timestep_range[1] > 0 && (timestep_range[1] < timestep_range[2] || timestep_range[2] == -1) && 
```

Comment:
you could move `timestep_range[2] = (timestep_range[2] == -1 ? ntimesteps : timestep_range[2])`
above and remove ` || timestep_range[2] == -1`

Reply:
Thanks, done.